web/admin/application/library/IvozProvider/Klear/Ghost/Carriers.php

```diff
@@ -9,12 +11,20 @@ class IvozProvider_Klear_Ghost_Carriers extends KlearMatrix_Model_Field_Ghost_Ab
      */
     private $fetchCallStats;
 
+    /**
+     * @var CarrierBalanceService
+     */
+    private $fetchCarrierBalance;
+
     public function __construct()
     {
         $serviceContainer = \Zend_Registry::get('container');
         $this->fetchCallStats = $serviceContainer->get(
             FetchCallStatsService::class
         );
+        $this->fetchCarrierBalance = $serviceContainer->get(
+            CarrierBalanceService::class
```

Comment:
We should use CompanyBalanceServiceInterface here.
You may need to set it public at library/CoreBundle/Resources/config/services/infraestructure/cgrates.yml

Reply:
Sure! Shouldn't the above service (FetchCallStats) also use an interface?

Reply:
Yes :-)